voltdbclient/node_conn.go

```diff
@@ -167,6 +162,15 @@ func (nc *nodeConn) listen(reader io.Reader, responseCh chan<- *bytes.Buffer) {
 			// the owner needs to reconnect
 			return
 		}
+		buf := bytes.NewBuffer(b)
+		s.SetReader(buf)
```

Comment:
@nshi you asked about the significance of `(*Decoder).SetReader` . here I'm using the same `*Decoder` on every iteration, by reseting the reader to read from the message buffer received.  I took inspiration from `*bytes.Reader` it is very convenient and useful.